### ISSUE TITLE
Exec command is not run at startup

### ISSUE THREAD

**Issue by JonasVerhofste** [hashicorp/consul-template#1617]
### Consul Template version
`v0.29.1`

### Configuration
(it's json with a trailing comma, but hcl apparently likes that just fine)
```hcl
"template": [
  {
    "destination": "/path/to/my/file",
    "exec": {
        "command": [
            "/etc/consul-ipvs/reload-ipvs.sh",
            "/etc/consul-ipvs/ipvsadm.conf"
        ]
    "perms": "0644",
    "source": "/path/to/my/template.ctmpl"
},
```

### Expected behavior

An earlier version of consul-template (v0.26.0) would run the exec-command at startup, regardless if the contents of the destination file changed or not. Current latest does not do this anymore. I'm unsure if the previous behaviour was a bug, but I can't find anything related to this behaviour in the changelog.

**Comment by eikenb** [comment]
Hey @JonasVerhofste, thanks for the report.

I'm trying to reproduce the behavior your describe in 0.26.0 and am unable to do it. With 0.26.0, if the template has fields w/o data it doesn't run the template's exec command (until the fields have data) and if the data is present so the template can render but the destination file is already present with the same contents it doesn't run it then either. It only runs the exec if the data is present and the output file either doesn't exist or exists with different content.

Running a consul server in dev mode with the following script, containing a simplified version of the problem, 0.26.0 works as you describe 0.29.1 does. Would you might looking this over and seeing if you can provide a similar simplified setup that can reproduce your issue. Thanks.

--

Note the command uses the old string format to work with 0.26.0.

```sh
#!/bin/sh

cat > foo.tmpl << EOF
: {{ key "foo" }}
EOF

consul kv put foo "hello world"

cat > conf.hcl << EOF
template {
    destination = "foo"
    exec {
        command = "/usr/bin/echo XXXXXXXXXXXXXXXXX"
    }
    source = "foo.tmpl"
}
EOF

consul-template \
    -config conf.hcl \
    -exec "tail -f foo" \
    -log-level debug
```

**Comment by JonasVerhofste** [comment]
Oh god, I think I just realised where my assumption went wrong, and it's completely unrelated to consul-template.
I fixed some issues with the way we handle the data in our consul-templates, and that might've fixed an issue that triggered the command in times where we _thought_ it didn't (and hence assumed it was an "exec at startup"). I however couldn't test those new templates with 0.26.0 as I started using some sprig functions, which weren't included in that version.

So this is probably indeed not a bug, but something we misinterpreted and made a wrong assumption on.
Appologies for taking up your time @eikenb!
